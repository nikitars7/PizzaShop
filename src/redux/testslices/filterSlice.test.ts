import filterReducer, {
  setCategoryId,
  setSortBy,
  setPageCount,
  setFilters,
  setSeatchValue,
  SortPropEnum,
} from "../slices/filterSlice";
const initialState = {
  searchValue: "",
  categoryId: 0,
  pageCount: 1,
  sort: {
    name: "rating",
    sortProperty: SortPropEnum.RATING_DESC,
  },
};
describe("filterSlice", () => {
  it("should return default state", () => {
    const result = filterReducer(undefined, { type: "" });
    expect(result).toEqual({
      searchValue: "",
      categoryId: 0,
      pageCount: 1,
      sort: {
        name: "rating",
        sortProperty: "rating",
      },
    });
    expect(result).toMatchSnapshot();
  });
  it("should setCategoryId value", () => {
    const action = {
      type: setCategoryId.type,
      payload: 1,
    };
    const result = filterReducer(initialState, action);
    const updatedState = { ...initialState, categoryId: 1 };
    expect(result).toEqual(updatedState);
    expect(result).toMatchSnapshot();
  });
  it("should setPageCount value", () => {
    const action = {
      type: setPageCount.type,
      payload: 2,
    };
    const result = filterReducer(initialState, action);
    const updatedState = { ...initialState, pageCount: 2 };
    expect(result).toEqual(updatedState);
    expect(result).toMatchSnapshot();
  });
  it("should setSearchValue value", () => {
    const action = {
      type: setSeatchValue.type,
      payload: "HelloWorld",
    };
    const result = filterReducer(initialState, action);
    const updatedState = { ...initialState, searchValue: "HelloWorld" };
    expect(result).toEqual(updatedState);
    expect(result).toMatchSnapshot();
  });
  it("should setSortBy value", () => {
    const action = {
      type: setSortBy.type,
      payload: {
        name: "price (ASC)",
        sortProperty: "-price",
      },
    };
    const result = filterReducer(initialState, action);
    const updatedState = {
      ...initialState,
      sort: {
        name: "price (ASC)",
        sortProperty: "-price",
      },
    };
    expect(result).toEqual(updatedState);
    expect(result).toMatchSnapshot();
  });
  it("should setFilters", () => {
    const action = {
      type: setFilters.type,
      payload: {
        categoryId: 1,
        sort: "-title",
        pageCount: 2,
        searchValue: "",
      },
    };
    const result = filterReducer(initialState, action);
    expect(result).toEqual({
      categoryId: 1,
      sort: "-title",
      pageCount: 2,
      searchValue: "",
    });
  });
});
