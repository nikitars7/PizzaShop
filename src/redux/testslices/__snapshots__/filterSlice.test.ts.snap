// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterSlice should return default state 1`] = `
Object {
  "categoryId": 0,
  "pageCount": 1,
  "searchValue": "",
  "sort": Object {
    "name": "rating",
    "sortProperty": "rating",
  },
}
`;

exports[`filterSlice should setCategoryId value 1`] = `
Object {
  "categoryId": 1,
  "pageCount": 1,
  "searchValue": "",
  "sort": Object {
    "name": "rating",
    "sortProperty": "rating",
  },
}
`;

exports[`filterSlice should setPageCount value 1`] = `
Object {
  "categoryId": 0,
  "pageCount": 2,
  "searchValue": "",
  "sort": Object {
    "name": "rating",
    "sortProperty": "rating",
  },
}
`;

exports[`filterSlice should setSearchValue value 1`] = `
Object {
  "categoryId": 0,
  "pageCount": 1,
  "searchValue": "HelloWorld",
  "sort": Object {
    "name": "rating",
    "sortProperty": "rating",
  },
}
`;

exports[`filterSlice should setSortBy value 1`] = `
Object {
  "categoryId": 0,
  "pageCount": 1,
  "searchValue": "",
  "sort": Object {
    "name": "price (ASC)",
    "sortProperty": "-price",
  },
}
`;
